processes/ingest/hathi_trust.py

```diff
@@ -99,23 +107,14 @@ def importFromHathiFile(self, hathiURL):
             self.readHathiFile(hathiTSV)
 
     def readHathiFile(self, hathiTSV):
-        processed_record_count = 0
-
-        while True:
-            if self.ingest_limit and processed_record_count >= self.ingest_limit:
+        for number_of_books_ingested, book in enumerate(hathiTSV):
+            if self.ingest_limit and number_of_books_ingested > self.ingest_limit:
                 break
 
-            try:
-                row = next(hathiTSV)
-                logger.info('Reading row')
-            except csv.Error as e:
-                logger.warning('Unable to read TSV row')
-                logger.debug(e)
-                continue
-            except StopIteration:
-                logger.info('Reached end of TSV file')
-                break
+            book_right = book[2]
+            book_date_updated = book[14]
+            start_date_time = datetime.now(timezone.utc).replace(tzinfo=None) - timedelta(days=1) 
```

Comment:
There could be some edge cases here where the start_date_time slightly changes as we are enumerating over the TSV.  I recommend setting it once above and passing it through the functions as a param or setting it as a class variable.

Reply:
Also we should only set the start_date_time and compare each book updated at if it's a daily run.

Reply:
That makes sense to to consider that case with the start_data_time changing so I will add it as a variable that's passed on to other functions. For your second comment, does this code line in the importFromHathiTrustDataFile not accomplish this comparison?

```
if hathi_date_modified > start_date_time:
                        self.importFromHathiFile(hathiFile['url'])
                        break
```

Reply:
It does for sure but we can pass that start_date_time into importFromHathiFile